src/BuildScriptGenerator/DotNetCore/DefaultAspNetCoreWebAppProjectFileProvider.cs

```diff
@@ -32,49 +33,47 @@ internal class DefaultAspNetCoreWebAppProjectFileProvider : IAspNetCoreWebAppPro
             _logger = logger;
         }
 
-        public string GetProjectFile(ISourceRepo sourceRepo)
+        public string GetRelativePathToProjectFile(ISourceRepo sourceRepo)
         {
             if (_probedForProjectFile)
             {
-                return _projectFile;
+                return _projectFileRelativePath;
             }
 
             var projectEnvVariablePath = _options.Project;
 
             string projectFile = null;
-            if (string.IsNullOrEmpty(projectEnvVariablePath))
+            if (!string.IsNullOrEmpty(projectEnvVariablePath))
             {
-                // Check if root of the repo has a .csproj or a .fsproj file
-                projectFile = GetProjectFileAtRoot(sourceRepo, DotnetCoreConstants.CSharpProjectFileExtension) ??
-                    GetProjectFileAtRoot(sourceRepo, DotnetCoreConstants.FSharpProjectFileExtension);
-            }
-            else
-            {
-                projectEnvVariablePath = projectEnvVariablePath.Trim();
-                projectFile = Path.Combine(sourceRepo.RootPath, projectEnvVariablePath);
+                var projectFileWithRelativePath = projectEnvVariablePath.Trim();
+                projectFile = Path.Combine(sourceRepo.RootPath, projectFileWithRelativePath);
                 if (!sourceRepo.FileExists(projectFile))
                 {
                     _logger.LogWarning($"Could not find the project file '{projectFile}'.");
                     throw new InvalidUsageException(
                         $"Could not find the project file '{projectFile}' specified by the environment variable" +
-                        $" '{EnvironmentSettingsKeys.Project}' with value '{projectEnvVariablePath}'. " +
+                        $" '{EnvironmentSettingsKeys.Project}' with value '{projectFileWithRelativePath}'. " +
                         "Make sure the path to the project file is relative to the root of the repo. " +
                         "For example: PROJECT=src/Dashboard/Dashboard.csproj");
                 }
 
                 // NOTE: Do not check if the project file specified by the end user is a web application since this
                 // can be a escape hatch for end users if our logic to determine a web app is incorrect.
-                return projectFile;
+                return projectFileWithRelativePath;
             }
```

Comment:
nit: I'd leave the `else` here to make it clearer that this is only when options.Project is null